src/engine/runtime.js

```diff
@@ -1218,29 +1222,52 @@ class Runtime extends EventEmitter {
             argJSON.check = argTypeInfo.check;
         }
 
-        const shadowType = (argInfo.menu ?
-            this._makeExtensionMenuId(argInfo.menu, context.categoryInfo.id) :
-            argTypeInfo.shadowType);
-        const fieldType = argInfo.menu || argTypeInfo.fieldType;
+        let valueName;
+        let shadowType;
+        let fieldName;
+        if (argInfo.menu) {
+            const menuInfo = argInfo.menu && context.categoryInfo.menuInfo[argInfo.menu];
```

Comment:
We're already in an if for `argInfo.menu` so we don't need the additional check here, right?
```suggestion
            const menuInfo = context.categoryInfo.menuInfo[argInfo.menu];
```